src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,381 @@
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            quitTo = data;
+        } else {
+            revert("RwaOutputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /**
+     * @notice Sets `who` address as the recipient.
+     * @param who Recipient address.
+     * @dev `who` address should have been whitelisted using `kiss`.
+     */
+    function pick(address who) external {
+        require(can[msg.sender] == 1, "RwaOutputConduit3/not-operator");
+        require(bud[who] == 1 || who == address(0), "RwaOutputConduit3/not-bud");
+        to = who;
+        emit Pick(who);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap DAI contract balance to GEM through PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     */
+    function push() external isMate {
+        _doPush(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external isMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external isMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out any GEM balance to `usr` address.
+     * @param usr Destination address
+     * @dev Can be called only by admin.
+     */
+    function yank(address usr) external auth {
+        uint256 amt = gem.balanceOf(address(this));
+        gem.transfer(usr, amt);
+        emit Yank(usr, amt);
+    }
+
+    /**
+     * @notice Calculate expected amount of GEM received for swapping `wad` amount of DAI.
+     * @param wad DAI amount.
+     * @return gemAmt Amount of GEM expected.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256) {
+        return mul(wad, WAD) / add(WAD, psm.tout()) / to18ConvertionFactor;
```

Comment:
The math is correct, however I think is a bit less prune to rounding errors to do:
```
mul(wad, WAD) / mul(add(WAD, psm.tout()), to18ConvertionFactor);
```

As there would be only one division operation.

Reply:
Nice catch, will update it in a moment

Reply:
Done